### ISSUE TITLE
Main TeX file without \begin{document}

### ISSUE THREAD

**Issue by dpo** [James-Yu/Atom-LaTeX#174]
When writing a Beamer presentation, I sometimes find it useful to produce handouts. I do that by creating a main TeX file (say `main.tex`) containing
```tex
\documentclass{beamer}
\input{body}
```
and a second main TeX file (say `main-handout.tex`) that I can compile later containing
```tex
\documentclass[xcolor=svgnames,handout,gray]{beamer}
\input{body}
```
The file `body.tex` contains the presentation itself, sets the title, contains `\begin{document} ... \end{document}`, etc.

Now I simply can't get Atom-LaTeX to recognize that `main.tex` is the main file. It insists on selecting `body.tex` as the main file and, of course, compilation fails. Even if I invoke the `build here` command. Is this happening because `main.tex` doesn't contain `\begin{document} ... \end{document}`? Is there a fix / workaround?

**Comment by dpo** [comment]
Well I should RTFM. Setting up a `.latexcfg` file does the trick. I would have expected `build here` to set the root file though.

**Comment by ashthespy** [comment]
Hmm, `build here` like the name suggests, attempts to build the active document. 
For your use case, I'd suggest the magic commands. I.e set `% !TEX root = main.tex` and `% !TEX root = main-handout.tex` respectively and then use the `LaTeX Build` command, and not build here :-)

**Comment by dpo** [comment]
Only I can't do that because I would need to add `% !TEX root` in `body.tex`, and it has two "TEX roots".

**Comment by ashthespy** [comment]
Ah yes, then if you hit build from your `body.tex` it won't know what to build. So `.latexcfg` it is!

**Comment by dpo** [comment]
Sorry but I'm still having trouble with this. I restarted Atom after upgrading to 1.30.0 and now even my `.latex.cfg` is ignored. I wonder if anyone is able to reproduce this. I have to say that even `.latex.cfg` isn't an ideal solution as I'd have to modify it to build the handouts. `Build here` is ideal but unfortunately, ignores my main file on grounds (I believe) that it doesn't contain `\begin{document}`.

**Comment by ashthespy** [comment]
@dpo I just whipped up a minimal experiment, and my suggestion from  https://github.com/James-Yu/Atom-LaTeX/issues/174#issuecomment-420574041 seems to work? aka just define magic comments in your `main.tex` and `main-handout.tex` setting their own filenames as root. 

But then - this would mean that any `Build` invocation from `body.tex` will just use the root file set by your last build.

**Comment by dpo** [comment]
It seems like a bug to me that `build here` refuses to set the root file.

**Comment by ashthespy** [comment]
Build here attempts to set the root file in the usual way with [`findMainSequence: (here)`](https://github.com/James-Yu/Atom-LaTeX/blob/master/lib/manager.coffee#L78)
I.e it looks for Magic commands or `\begin{document}` tags. IMO the normal use case for `build here` is to build the current open editor as the root file. 

In your current example as you have neither of these two in your `root` file, it fails. How would you otherwise propose that `build here` determines the root file for your use case?

**Comment by dpo** [comment]
> IMO the normal use case for build here is to build the current open editor as the root file.

Yes, I agree. What I'm saying is that if I use `build here` when `main.tex` is the active editor, Atom-LaTeX still insists on building `body.tex` as the main file (presumably because it contains the `\begin{document}`).

I can organize my files differently if this is too much trouble, but I'm thinking `build here` may be trying to be too smart when in reality, it should just build **here**. I understand that `build` tries to be smart. I would expect` build here` to simply take the current active editor and pass that file straight to `latexmk`. If changing that behavior disturbs anything else, perhaps there could be another command, say, `build HERE!`.

**Comment by ashthespy** [comment]
While I see your point about it trying to be an eager beaver to determine the root file, might I ask why the proposed magic commands wouldn't work in your scenario? That shouldn't require any major restructuring/config changes for your use case?

We could add a case of attempting building the current editor if the current two checks fail - but I shall leave it as a future enhancement - but PR's are welcome!

**Comment by dpo** [comment]
Thanks. I'll have to test the magic commands you suggest, but the `.latexcfg` route failed for me yesterday after a restart/upgrade of Atom.

**Comment by dpo** [comment]
The magic commands appear to work for me. Thanks for the suggestion!
